### DISCUSSION TITLE
Catching errors from `useLiveQuery`

### DISCUSSION THREAD

**Discussion by nkalupahana** [dexie/Dexie.js#2031] [Q&A] [answered]
Because of [this Safari WebKit IndexedDB bug](https://bugs.webkit.org/show_bug.cgi?id=273827), IndexedDB sometimes fails to open on user's devices. When this happens, the `useLiveQuery` [call on my website](https://github.com/nkalupahana/baseline/blob/main/src/components/Journal/FinishJournal.jsx#L31) fails, which causes the entire component to fail rendering, making the entire website look broken. 

Is there any way to catch an error like this when using `useLiveQuery`, or will I have to switch to calling Dexie inline with a try-catch or something like that, in order to avoid the failing Dexie call from breaking the entire component? Thanks!

**Comment by dfahlander** [comment, marked as answer]
Use an [error boundary](https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary) to catch and inspect any error happening in live queries.

**Reply by maidi29** [reply]
@nkalupahana were you able to derive if the errors catched in error boundary were from `useLiveQuery`?

**Reply by nkalupahana** [reply]
I didn't try, sorry. Probably worth creating a new discussion with your specific use case so people will see it :)

Here is how I used an error boundary, for anyone else who ends up here: https://github.com/nkalupahana/baseline/pull/414/changes (in relation to https://bugs.webkit.org/show_bug.cgi?id=277615)
